### ISSUE TITLE
Histogram Binning Estimator

### ISSUE THREAD

**Issue by ojjj13** [metabase/metabase#37118]
Current binning method for histogram plot is fixed with few options, the default "AUTO" amount of bins does not always gives a proper outcome (in most case there are too few bins).

I wonder if it is possible to implement a estimator for the number of bins like what matplotlib does for their "AUTO" binning method: 

[https://numpy.org/doc/stable/reference/generated/numpy.histogram_bin_edges.html#numpy.histogram_bin_edges]

The result of such implementation will likely to give a more meaningful histogram compared to the existed one, it is also a imporvement to the insight page which the bins in there are usually not enough for a meaningful implication.

**Comment by paoliniluis** [comment]
@ojjj13 is this something like https://github.com/metabase/metabase/issues/9321 or https://github.com/metabase/metabase/issues/12847 ?

**Comment by ojjj13** [comment]
@paoliniluis No, I was not referring to a customize bin size, but an estimator for bin size using similar algorithm like the matplotlib auto option, which is the maximum of the ‘sturges’ and ‘fd’ estimators, the algorithm is provided in the link in the original post.

**Comment by bbougon** [comment]
Plus one, it would be great to define the following functions in order to have a better display of bins :

- ‘fd’ (Freedman Diaconis Estimator)
   Robust (resilient to outliers) estimator that takes into account data variability and data size.
- ‘doane’
   An improved version of Sturges’ estimator that works better with non-normal datasets.
- ‘scott’
   Less robust estimator that takes into account data variability and data size.
- ‘stone’
   Estimator based on leave-one-out cross-validation estimate of the integrated squared error. Can be regarded as a generalization of Scott’s rule.
- ‘rice’
   Estimator does not take variability into account, only data size. Commonly overestimates number of bins required.
- ‘sturges’
   R’s default method, only accounts for data size. Only optimal for gaussian data and underestimates number of bins for large non-gaussian datasets.

In pryority, it woujd be great to have 
1. Freedman Diaconis
2. Doane
3. Sturges

Thank you 🙏
